include/deal.II/lac/linear_operator.h

```diff
@@ -722,6 +723,72 @@ inverse_operator(const LinearOperator<Range, Domain, Payload> &op,
   return return_op;
 }
 
+
+/**
+ * @relatesalso LinearOperator
+ *
+ * Variant of above function that takes an rvalue-reference as
+ * preconditioner argument. This function takes care of instances where
+ * either a LinearOperator is created as a temporary argument, or where for
+ * example <code>PreconditionIdentity()</code> is supplied.
+ *
+ * In this case, instead of storing a reference to the preconditioner as is
+ * done in the previous function, we store a full copy of the
+ * preconditioner instead.
+ *
+ * @ingroup LAOperators
+ */
+template <typename Payload,
+          typename Solver,
+          typename Preconditioner,
+          typename Range  = typename Solver::vector_type,
+          typename Domain = Range,
+          typename        = typename std::enable_if<
+            !std::is_lvalue_reference<Preconditioner>::value>::type>
+LinearOperator<Domain, Range, Payload>
+inverse_operator(const LinearOperator<Range, Domain, Payload> &op,
+                 Solver &                                      solver,
+                 Preconditioner &&                             preconditioner)
```

Comment:
In the text below, can you point out where the copy actually happens? Some documentation of this issue within the code would be nice :-)

Reply:
I have removed the variant altogether.